### PR TITLE
Fix vincenty inverse for equatorial line

### PR DESCRIPTION
**Pull request by Brtle**
I looked in detail the calculation of vincenty inverse to fix the unit test on Alpine (see #1825). I think that a case is not correctly handled when `cos²α = 0`. On [this website](https://www.movable-type.co.uk/scripts/latlong-vincenty.html), a note describes this case:

> Vincenty observes that eqn. (18) becomes indeterminate over equatorial lines (since cos²α → 0); in this case, set cos2σm to 0 and the result is computed correctly.

The problematic unit test calculates the distance between (0, 0) and (0, 17) which is precisely an equatorial line. Handling the `cos²α = 0` fixes this issue. Another test on the equatorial line was expected to fail (distance between (0, 0) and (0, 13)) and is converging now.

Moreover, this case is also handled in [geopy](https://github.com/geopy/geopy/blob/master/geopy/distance.py#L573) in the same way.

I also did some minor changes:

- Some cos/sin was already calculated, I reused directly the value instead of recalculating the cos/sin.
- `α` was calculated but only `cos²α` was useful, I calculated it directly without using `asin`.
- This code block was inside the loop for no reason:

```python
u2 = pow(math.cos(alpha), 2) * (a * a - b * b) / (b * b)
_a = 1 + (u2 / 16384) * (4096 + u2 * (-768 + u2 *
                                        (320 - 175 * u2)))
_b = (u2 / 1024) * (256 + u2 * (-128 + u2 * (74 - 47 * u2)))
delta_sigma = _b * sin_sigma * \
    (cos2sigma_m + (_b / 4) *
        (cos_sigma * (-1 + 2 * pow(cos2sigma_m, 2)) - (_b / 6) *
            cos2sigma_m * (-3 + 4 * sqr_sin_sigma) *
            (-3 + 4 * pow(cos2sigma_m, 2))))

dist = b * _a * (sigma - delta_sigma)
alpha12 = math.atan2(
    (math.cos(u_2) * math.sin(dlon)),
    (math.cos(u_1) * math.sin(u_2) -
        math.sin(u_1) * math.cos(u_2) * math.cos(dlon)))
alpha21 = math.atan2(
    (math.cos(u_1) * math.sin(dlon)),
    (-math.sin(u_1) * math.cos(u_2) +
        math.cos(u_1) * math.sin(u_2) * math.cos(dlon)))
```

It was useless to calculate these variables at each iteration, I moved the block at the end of the function.

### CODE REVIEW

File: obspy/geodetics/base.py [hunk 1]
```diff
@@ -140,41 +147,29 @@ def calc_vincenty_inverse(lat1, lon1, lat2, lon2, a=WGS84_A, f=WGS84_F):
...
+            sqr_cos_alpha = 1 - sin_alpha * sin_alpha
+            if _isclose(sqr_cos_alpha, 0):
+                # Equatorial line
+                cos2sigma_m = 0
+            else:
+                cos2sigma_m = cos_sigma - \
+                    (2 * math.sin(u_1) * math.sin(u_2) / sqr_cos_alpha)
+
+            c = (f / 16) * sqr_cos_alpha * \
+                (4 + f * (4 - 3 * sqr_cos_alpha))
```
Comment (QuLogic) [inline obspy/geodetics/base.py]: I think this fits on one line now?

Comment (Brtle) [reply obspy/geodetics/base.py]: Fixed!

File: obspy/geodetics/base.py [hunk 2]
```diff
@@ -40,6 +40,13 @@
 WGS84_F = 1 / 298.257223563
 
 
+def _isclose(a, b, rel_tol=1e-09, abs_tol=0.0):
```
[the remark this answers is not in the document]
Comment (Brtle) [reply obspy/geodetics/base.py]: According to [this comment](https://github.com/obspy/obspy/pull/2265#issuecomment-448998882), `np.isclose` is not available for numpy < 1.7. That's why I didn't use it.

**Comment by Brtle** [comment]
> Looks good, thanks! Can you add a one-liner to the changelog as well?

Oops, I forgot that. Done!

**Comment by megies** [comment]
Thanks @Brtle! :tada:

**Pull request merged**